corehq/apps/app_manager/tests/test_suite.py

```diff
@@ -89,6 +92,7 @@ def test_owner_name(self, *args):
     def test_printing(self, *args):
         self._test_generic_suite('app_print_detail', 'suite-print-detail')
 
+    @privilege_enabled(privileges.APP_DEPENDENCIES)
```

Comment:
Curious why this was required ?

Reply:
Reasonable question @ajeety4 and well spotted.

Actually this test (and other ones) have been written using `SimpleTestCase` class which prohibits or is meant to not run DB queries.
The change I made for the feature, to check on privilege, fires up SQL queries, which caused tests to fail.

So, the option was to either change `SimpleTestCase` to `TestCase` or to mock the privilege check. I chose the latter to keep test db-independent and also because I would have done the patch anyway to let the test use app dependency feature by default.

Reply:
Looking at the change 

https://github.com/dimagi/commcare-hq/pull/35237/files#diff-db20e55a79b6ed9c4bfbaa81acc7318970ec4634132220838b7abadb52379899R558

the tests do appear to have run with the toggle check. I understand that feature flags are also in DB but that still worked fine due to some unknown reason which I am okay to not dive into right now.

Reply:
Noted. Thanks Manish for the clarification.